virtual_ecosystem/models/animal/functional_group.py

```diff
@@ -129,3 +144,24 @@ def import_functional_groups(
     ]
 
     return functional_group_list
+
+
+def get_functional_group_by_name(
+    functional_groups: tuple[FunctionalGroup, ...], name: str
+) -> FunctionalGroup:
+    """Retrieve a FunctionalGroup by its name from a tuple of FunctionalGroup instances.
+
+    Args:
+        functional_groups: Tuple of FunctionalGroup instances.
+        name: The name of the FunctionalGroup to retrieve.
+
+    Returns:
+        The FunctionalGroup instance with the matching name.
+
+    Raises:
+        ValueError: If no FunctionalGroup with the given name is found.
+    """
+    for fg in functional_groups:
+        if fg.name == name:
+            return fg
```

Comment:
I guess there will be only one (or none) groups with the given name, but not more than one, right?

Reply:
Each functional group has a unique name. It's not clear yet what the final format of the naming system will be but I expect to maintain uniqueness.